ui/src/pages/users/list/invite-users.tsx

```diff
@@ -0,0 +1,275 @@
+import { useMemo, useState, useEffect } from "react";
+import {
+  Button,
+  Dialog,
+  Flex,
+  Label,
+  Select,
+  Text,
+  TextArea,
+  toast,
+} from "@raystack/apsara/v1";
+import { PlusIcon } from "@radix-ui/react-icons";
+import { AxiosError } from "axios";
+import * as z from "zod";
+import { Controller, useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { api } from "~/api";
+import { SearchOrganizationsResponseOrganizationResult } from "~/api/frontier";
+import { SCOPES, DEFAULT_ROLES } from "~/utils/constants";
+import styles from "./invite-users.module.css";
+
+const inviteSchema = z.object({
+  role: z.string(),
+  organizationId: z.string(),
+  emails: z
+    .string()
+    .transform(value => value.split(",").map(str => str.trim()))
+    .pipe(z.array(z.string().email())),
+});
+
+type InviteSchemaType = z.infer<typeof inviteSchema>;
+
+const getAllOrganizations = async () => {
+  return api
+    .adminServiceSearchOrganizations({})
+    .then(res => res.data.organizations ?? []);
+};
+
+const getDefaultRoles = async () => {
+  return api
+    .frontierServiceListRoles({
+      scopes: [SCOPES.ORG],
+    })
+    .then(res => res.data?.roles ?? []);
+};
+
+export const InviteUser = () => {
+  const [open, onOpenChange] = useState(false);
+  const [organizations, setOrganizations] = useState<
+    SearchOrganizationsResponseOrganizationResult[]
+  >([]);
+  const [roles, setRoles] = useState<any[]>([]);
+  const [isRolesLoading, setIsRolesLoading] = useState(false);
+  const [isOrgsLoading, setIsOrgsLoading] = useState(false);
+
+  const defaultRoleId = useMemo(
+    () => roles?.find(role => role.name === DEFAULT_ROLES.ORG_VIEWER)?.id,
+    [roles],
+  );
+
+  useEffect(() => {
+    setIsOrgsLoading(true);
+    setIsRolesLoading(true);
+
+    getAllOrganizations()
+      .then(data => {
+        setOrganizations(data);
+      })
+      .catch(error => {
+        console.error(error);
+      })
+      .finally(() => {
+        setIsOrgsLoading(false);
+      });
+
+    getDefaultRoles()
```

Comment:
Let's use async-await here.
Let's not use different types of async handling in the same file.